hoprd/rest-api/src/peers.rs

```diff
@@ -61,21 +64,27 @@ pub(crate) struct PeerIdParams {
     tag = "Peers",
 )]
 pub(super) async fn show_peer_info(
-    Path(PeerIdParams { peer_id }): Path<PeerIdParams>,
+    Path(DestinationParams { destination }): Path<DestinationParams>,
     State(state): State<Arc<InternalState>>,
 ) -> impl IntoResponse {
     let hopr = state.hopr.clone();
-    match PeerId::from_str(peer_id.as_str()) {
-        Ok(peer) => (
-            StatusCode::OK,
-            Json(NodePeerInfoResponse {
-                announced: hopr.multiaddresses_announced_on_chain(&peer).await,
-                observed: hopr.network_observed_multiaddresses(&peer).await,
-            }),
-        )
-            .into_response(),
-        Err(_) => (StatusCode::BAD_REQUEST, ApiErrorStatus::InvalidPeerId).into_response(),
-    }
+
+    let destination = destination.fulfill(hopr.peer_resolver()).await;
+    let peer_id = match destination {
+        Ok(destination) => match destination.peer_id {
```

Comment:
The nested `match` can be flattened:

````rust

match destination {
   Ok(Some(dst)) => ...,
   Ok(None) => ...,
   Err(e) => ...,
}

````

This applies probably to multiple places where you nest the `match`.